docs/design.md

```diff
@@ -18,7 +18,7 @@ Firecracker virtual machine manager (VMM).
 
 1. Firecracker can safely run workloads from different customers on the same
    machine.
-1. Customers can create microVMs with any combination of vCPU and memory to
+1. Customers can create microVMs with any combination of vCPU (up to 32) and memory to
```

Comment:
I'm not sure a clarification on operational limitations/constraints should be put in the design document unless its a constraint that exists *because* of the design. My 2c.

Reply:
While I agree that having this information together with the design may be convoluted to be found, in this document we are specifying many other features/capabilities which we support because we do not have at the moment even any other better place where to do so (SPECIFICATION.md in reality state our promises in terms of performance more than features/capability).  Currently the limit comes from our design in Firecracker or better we imposed a soft limit of 32 [here](https://github.com/firecracker-microvm/firecracker/blob/1a61cbafa9e1dda6aef92b94f5beaffcd0c296ff/src/vmm/src/vmm_config/machine_config.rs#L11)

Reply:
>  Currently the limit comes from our design in Firecracker
I don't agree that the limit comes from the design, but from the implementation.

> ... in this document we are specifying many other features/capabilities which we support because we do not have at the moment even any other better place ...

This is not a great justification for placing it here. The documentation is quite tough to navigate, better/correct sensical placement can make a difference. If that place doesn't exist yet, it can still be created.

Reply:
> I don't agree that the limit comes from the design, but from the implementation.
That is what i meant but our implementation reflect the limitation which we have in Firecracker because with bigger vCPU numbers we have degradation which will make Firecraker not usable. At the same time, this simple modification of that sentence is not claiming that a customer can use up to 32 vCPU due to the design and i can make it maybe explicit changing the bracket into "(with a soft limit up to 32)". design.md is the name of the file but in that file there is much more. Just read the first sentence "_The scope of this document is to describe the features and architecture of the Firecracker virtual machine manager (VMM)._"

> The documentation is quite tough to navigate, better/correct sensical placement can make a difference.
I do not think the documentation is tough to navigate or this change will affect the readability. Actually this file is well structured and present a section on Features that we intended to include in Firecracker. We can of course in future think about a separated file with just a series of features which we want to highlight and their corresponding values, but this to me still seem a valid incremental step in the direction to provide the max amount of information to our users now while may undertake in future the task of creating a specific page (which i guess is your current suggestion?)  and which need even more thoughts in order to finalize which other parameters need to be included.